catalyst-gateway/bin/src/service/api/health/ready_get.rs

```diff
@@ -35,5 +46,48 @@ pub(crate) type AllResponses = WithErrorResponses<Responses>;
 /// service that are ready.
 #[allow(clippy::unused_async)]
 pub(crate) async fn endpoint() -> AllResponses {
-    Responses::NoContent.into()
+    // Check Event DB connection
+    let event_db_live = event_db_is_live();
+
+    // When check fails, attempt to re-connect
+    if !event_db_live {
+        establish_connection();
+        // Re-check, if success, enable flag.
+        if EventDB::connection_is_ok() {
+            set_event_db_liveness(true);
+        }
+    };
+
+    // Check Index DB connection
+    let index_db_live = index_db_is_live();
+
+    // When check fails, attempt to re-connect
+    if !index_db_live {
+        CassandraSession::init();
+        // Re-check, if success, enable flag.
+        if CassandraSession::wait_until_ready(core::time::Duration::from_secs(1), false)
+            .await
+            .is_ok()
+        {
+            set_index_db_liveness(true);
+        }
+    }
+
+    let success_response = Responses::NoContent.into();
+
+    // Return 204 response if check passed initially.
+    if index_db_live && event_db_live {
+        return success_response;
+    }
+
+    // Otherwise, re-check, and return 403 response if all is good.
+    if index_db_is_live() && event_db_is_live() {
+        return success_response;
+    }
```

Comment:
I assume that the 403 comment is wrong?
If yes can we just do

```
 if index_db_is_live() && event_db_is_live() {
        return success_response;
    }
```

Reply:
Yup, the comment is wrong. Thanks.